var/spack/repos/builtin/packages/charliecloud/package.py

```diff
@@ -36,6 +36,12 @@ class Charliecloud(AutotoolsPackage):
     # See https://github.com/spack/spack/pull/16049.
     conflicts('platform=darwin', msg='This package does not build on macOS')
 
+    # See https://github.com/spack/spack/issues/19310
+    conflicts('^py-sphinx-rtd-theme@0.5.0', when='%gcc@:4.8',
+              msg="py-sphinx-rtd-theme@0.5.0 fails to build with gcc 4.8 "
+                  "(see https://github.com/spack/spack/issues/19310); specify "
+                  + "an older version, e.g., py-sphinx-rtd-theme@0.4.3")
```

Comment:
This is actually in the wrong place.  The package that has a conflict with `gcc@:4.8` is `node-js`.  Can you add this conflict there?

I tried this out with the new concretizer (which will be experimentally available in the November 0.16 release) to see if we can handle it.  If I specify that `conflicts("%gcc@:4.8")` in `node-js`, the new concretizer correctly solves around it by either using a newer compiler for `node-js` (this was its first choice) or by downgrading `py-sphinx-rtd-theme` to `0.4.3` if I rule out compiler changes.

I think you can leave this conflict in because it'll give the user a better error message for now, and it'll tell them what they need to do to get things working.  But can you add a note saying that this is only here to help users until the concretizer is improved?

Reply:
Will do, thanks!